src/targets/gpu/include/migraphx/gpu/context.hpp

```diff
@@ -274,6 +276,24 @@ struct context
         this->current_device = std::make_shared<hip_device>(0, n_streams);
     }
 
+    void wait_for(any_ptr queue)
+    {
+        auto status = hipEventRecord(begin_event.get(), queue.get<hipStream_t>());
+        if(status != hipSuccess)
+            MIGRAPHX_THROW("failed to record " + hip_error(status));
+
+        get_stream().wait(begin_event.get());
```

Comment:
```      
      set_current_device();
      auto status = hipStreamWaitEvent(queue.get(), event, 0);
```

Reply:
Similar to above, should I be not reusing the existing blocks from the current stream in this context?

Reply:
`wait_for` is meant for `get_stream()` to wait on the user-queue, which means we would wait on `get_stream()` and record on the user-queue.